### ISSUE TITLE
[CI-fail] RemoteSpeedTest.py on macOS and Linux (Windows passes)

### ISSUE THREAD

**Issue by Dschoni** [pyqtgraph/pyqtgraph#881]
On TravisCI remote plotting fails on python 2.7 with pyqt 4 with the following stacktrace:

```
Python 2.7.16 |Anaconda, Inc.| (default, Mar 14 2019, 21:00:58) 
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> >>> ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... Traceback (most recent call last):
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 919, in __call__
    return self._handler.callObj(obj=self, args=args, kwds=kwds, **opts)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 564, in callObj
    return self.send(request='callObj', opts=dict(obj=obj, args=args, kwds=kwds), byteData=byteMsgs, **opts)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 469, in send
    return req.result()
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 637, in result
    while not self.hasResult():
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 655, in hasResult
    self.result(block=False)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 648, in result
    self._result = self.proc.getResult(self.reqId)  ## raises NoResultError if result is not available yet
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 486, in getResult
    self.processRequests()
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/processes.py", line 371, in processRequests
    RemoteEventHandler.processRequests(self)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 126, in processRequests
    while self.conn.poll():
IOError: poll() gave POLLNVAL or POLLERR
Traceback (most recent call last):
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/processes.py", line 371, in processRequests
    RemoteEventHandler.processRequests(self)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 126, in processRequests
    while self.conn.poll():
IOError: poll() gave POLLNVAL or POLLERR
Traceback (most recent call last):
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 919, in __call__
    return self._handler.callObj(obj=self, args=args, kwds=kwds, **opts)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 564, in callObj
    return self.send(request='callObj', opts=dict(obj=obj, args=args, kwds=kwds), byteData=byteMsgs, **opts)
  File "/home/travis/repo-clone/pyqtgraph/multiprocess/remoteproxy.py", line 452, in send
    self.conn.send(request)
IOError: [Errno 104] Connection reset by peer
```

On a locally installed python 2.7 with pyqt 4 (anaconda) the error could not be reproduced.

**Comment by Dschoni** [comment]
Same is true for Azure CI

**Comment by danielhrisca** [comment]
How about dropping support for Pyhon. 3.6? This would open the possibility to use many nice features that were intriduced in Pyhton 3.6

I image that you probably want to make anither release with the support enabled since the last release was so far ago.

**Comment by j9ac9k** [comment]
> How about dropping support for Pyhon. 3.6? This would open the possibility to use many nice features that were intriduced in Pyhton 3.6
> 
> I image that you probably want to make anither release with the support enabled since the last release was so far ago.

You mean drop support for python 2.7?  We've talked about that, and it's support will be going away/phased out, but not for this coming release.

I also have no reason to think that this is a python2 issue, the test suite only tests examples when installed against pyside1/pyqt4 bindings, which mostly are installed against python2 environments.

**Comment by danielhrisca** [comment]
I mean drop support for python <3.6 for the next release

**Comment by j9ac9k** [comment]
not going to happen for this coming release, pyside2-5.12/py27 is a valid install environment... we've had some discussions about dropping python2 support, but while new qt bindings support python 2.7, pyqtgraph should likely support it too.

**Comment by j9ac9k** [comment]
```
Remote Plotting...................................FAILED


Python 3.7.3 (default, Apr 16 2019, 09:37:09) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> >>> ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... ... QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-vsts'
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-vsts'
Traceback (most recent call last):
  File "/home/vsts/work/1/s/pyqtgraph/multiprocess/remoteproxy.py", line 919, in __call__
    return self._handler.callObj(obj=self, args=args, kwds=kwds, **opts)
  File "/home/vsts/work/1/s/pyqtgraph/multiprocess/remoteproxy.py", line 564, in callObj
    return self.send(request='callObj', opts=dict(obj=obj, args=args, kwds=kwds), byteData=byteMsgs, **opts)
  File "/home/vsts/work/1/s/pyqtgraph/multiprocess/remoteproxy.py", line 469, in send
    return req.result()
  File "/home/vsts/work/1/s/pyqtgraph/multiprocess/remoteproxy.py", line 639, in result
    raise ClosedError()
pyqtgraph.multiprocess.remoteproxy.ClosedError
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/vsts/work/1/s/pyqtgraph/widgets/RemoteGraphicsView.py", line 168, in close
    self.shm.close()
BufferError: cannot close exported pointers exist
```

Looks like this is failing on both linux/macOS builds, but funny enough, it runs okay on windows configurations.

**Comment by j9ac9k** [comment]
With #911 these failures are now showing up as actual test failures; (not just failing but saying passed).

**Comment by pijyoi** [comment]
The following error was fixed in #1517
>
```
Traceback (most recent call last):
  File "/home/vsts/work/1/s/pyqtgraph/widgets/RemoteGraphicsView.py", line 168, in close
    self.shm.close()
BufferError: cannot close exported pointers exist
```

From the commit message of https://github.com/pyqtgraph/pyqtgraph/pull/1517/commits/2f4ceee1ced5f49350626e42ad6203dfc9f13ee9
```
fix BufferError: cannot close exported pointers exist
for some reason, even though the ctypes object falls out of function
scope, it causes a lingering reference.
in any case, the use of ctypes is no longer necessary.
```

**Comment by j9ac9k** [comment]
Thanks @pijyoi  for flagging this issue for closing!
